### DISCUSSION TITLE
development environment and run the application locally

### DISCUSSION THREAD

**Discussion by Ali7040** [ruxailab/RUXAILAB#309] [Q&A] [answered]
@jvJUCA  I read your documentation on how to run the application locally, but Firebase doesn't provide a database URL. What should I do?

![Screenshot 2024-02-23 at 4 01 22 PM](https://github.com/ruxailab/RUXAILAB/assets/81979505/dad01901-e4f9-4d81-9818-7fcf5d2ec898)

**Comment by marcgc21** [comment]
Hi @Ali7040, remember that Firebase uses Firestore and Realtime Database for that sense. On our case we are using Firestore.
However, can you point us where did you find that you need thie database URL reference?

Att,

M.

**Reply by Ali7040** [reply]
@marcgc21  In the RUXAILAB repo docs. here is the screenshot check it out 
![Screenshot 2024-02-23 at 6 03 02 PM](https://github.com/ruxailab/RUXAILAB/assets/81979505/357b9a6b-a550-4830-9e82-56bd852167ca)

**Comment by jvJUCA** [comment]
Dear @Ali7040 ,

Here's a helpful setup guide for the project:

<a href='https://youtu.be/dAf4LRxITCc'>Tutorial - Running RUXAILAB with Firebase Emulators   🖥️</a>

Hope it's useful!

Best regards,

@jvJUCA

**Comment by pankaj87072** [comment]
hey @jvJUCA my exams are over and now i m giving my time to this org but after setting up local env.. 
![Screenshot from 2024-03-03 00-10-16](https://github.com/ruxailab/RUXAILAB/assets/112722707/4b4ada77-285b-436c-8b70-628cc49e4eb0)
my page is not loading

**Comment by pankaj87072** [comment]
hey @marcgc21 can you help pls..

**Reply by KarinePistili** [reply]
Hello @pankaj87072 , we've just published a video on the readme with a guide on how to setup the project locally using Firebase Emulators. Have you checked it out?

In case you did, and you still have the error, I believe there is something missing in your .env. Have you created your own Firebase project and got the information from there? Latest and stable code lives on the **develop** branch.

We would need further details to help you in this issue, such as the logs provided in the devtools for example.

I will wait for your response :)

**Reply by pankaj87072** [reply]
@KarinePistili yes i have checked it already and all the setups are done,all things are working properly but when i run npm run serve and follows the link it opens the blank page.

**Reply by pankaj87072** [reply]
i almost wasted full day on it

**Reply by KarinePistili** [reply]
Can you send the logs from devtools?

**Reply by pankaj87072** [reply]
Mar 04, 2024 7:36:14 PM com.google.cloud.datastore.emulator.firestore.websocket.WebSocketServer start
INFO: Started WebSocket server on ws://127.0.0.1:9150
API endpoint: http://127.0.0.1:8081
If you are using a library that supports the FIRESTORE_EMULATOR_HOST environment variable, run:

   export FIRESTORE_EMULATOR_HOST=127.0.0.1:8081

If you are running a Firestore in Datastore Mode project, run:

   export DATASTORE_EMULATOR_HOST=127.0.0.1:8081

Note: Support for Datastore Mode is in preview. If you encounter any bugs please file at https://github.com/firebase/firebase-tools/issues.
Dev App Server is now running.

Mar 04, 2024 7:36:31 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
INFO: Detected non-HTTP/2 connection.
Mar 04, 2024 7:36:31 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
INFO: Detected HTTP/2 connection.
Mar 04, 2024 7:47:21 PM com.google.cloud.datastore.emulator.firestore.websocket.WebSocketChannelHandler initChannel
INFO: Connected to new websocket client
*** shutting down gRPC server since JVM is shutting down
*** server shut down

**Reply by KarinePistili** [reply]
Hi I mean the devtools of the browser when you enter localhost:8080, the ruxailab url. Is this that? It looks differente from usual expected

**Reply by pankaj87072** [reply]
![Screenshot from 2024-03-05 00-18-59](https://github.com/ruxailab/RUXAILAB/assets/112722707/d1f3ad56-ee57-40f7-b9ad-56deb91578b5)

**Reply by pankaj87072** [reply]
it just showing the firebase api error and waiting even after multiple attempts of generating new api key and copying it to .env file

**Comment by pankaj87072** [comment]
srry for late reply

**Comment by priyakumari02** [comment]
Hello @KarinePistili ,
I followed the video tutorial , and after running npm  run serve 
In localhost i am getting a blank page
![image](https://github.com/ruxailab/RUXAILAB/assets/77202746/641c600d-6a36-4138-9da4-9c74366428d2)

**Reply by pankaj87072** [reply]
@priyakumari02 same with me

**Comment by priyakumari02** [comment]
And on running firebase enumalator:start i am getting this error
![Screenshot from 2024-03-05 07-43-36](https://github.com/ruxailab/RUXAILAB/assets/77202746/5cd964df-bab0-414b-882b-5928bade4a01)

**Reply by pankaj87072** [reply]
Have you used firebase init

**Reply by priyakumari02** [reply]
Yes

**Reply by pankaj87072** [reply]
Run firebase use --add and select the project

**Reply by KarinePistili** [reply]
We will try to recreate the errors you are having and will give a feedback soon

**Reply by pankaj87072** [reply]
@KarinePistili ohk mam

**Reply by priyakumari02** [reply]
OK @KarinePistili

**Reply by KarinePistili** [reply]
hello everybody, I've just put an answer that could help you on the discussion, check it out and tell me if it helps you ;)

**Comment by skushagra9** [comment]
maybe, something wrong with how variables are passed in  development mode, instead of calling the varibales from env, you can just paste the variables, and remove the conditional rendering, it will work

**Reply by skushagra9** [reply]
just create a db with firestore and it would work

**Comment by pankaj87072** [comment]
@KarinePistili hey same problem occur using github codesapce also pls review it 
![Screenshot from 2024-03-06 00-30-33](https://github.com/ruxailab/RUXAILAB/assets/112722707/a64146c4-0104-45a9-855e-dc9fd017743c)

**Reply by KarinePistili** [reply]
hello @pankaj87072 , I've just put a possible fix in my last answer, try to check it out and see if it works for you :)

**Comment by abhisek-1221** [comment]
- Emulators are running but getting blank page in port 8080 (vue localhost server)

![Screenshot 2024-03-06 at 4 36 32 PM](https://github.com/ruxailab/RUXAILAB/assets/110292494/a16e7ba6-051d-4041-89ee-c27db2e37540)
![Screenshot 2024-03-06 at 4 40 28 PM](https://github.com/ruxailab/RUXAILAB/assets/110292494/1387f789-712e-46f9-b3ab-3d5451cab35b)

@KarinePistili @marcgc21 Need help!

**Reply by KarinePistili** [reply]
hello @abhisek-1221 , try checking the info I've put in the answer below, see if it helps you

**Comment by KarinePistili** [comment, marked as answer]
There are some things you could check to see if it will work for you:

### 1. Check the name of your env vars, see if they match the following:
```
VUE_APP_FIREBASE_API_KEY_DEV=""
VUE_APP_FIREBASE_AUTH_DOMAIN_DEV=""
VUE_APP_FIREBASE_STORAGE_BUCKET_DEV=""
VUE_APP_FIREBASE_DB_URL_DEV=""
VUE_APP_FIREBASE_PROJECT_ID_DEV=""
VUE_APP_FIREBASE_MESSAGING_SENDER_ID_DEV=""
VUE_APP_FIREBASE_APP_ID_DEV=""
VUE_APP_I18N_LOCALE=""
VUE_APP_I18N_FALLBACK_LOCALE=""
LARAVEL_PDF=""
```

If you are using the development mode, they have the **_DEV** at the end. The screenshot below shows how the env vars are used on the code to initialize firebase project.

![image](https://github.com/ruxailab/RUXAILAB/assets/43211581/837dbd5b-5c1f-454b-ac29-ad44b29394fe)

### 2. Check if you correclty put the `VUE_APP_FIREBASE_DB_URL_DEV=""` variable. You need to create a database on firestore dashboard, this option will appear on project config after that.

**Reply by abhisek-1221** [reply]
Thanks @KarinePistili ! It's working

**Reply by DIWAKARKASHYAP** [reply]
thanks @KarinePis tili working
